docs/parse_argparse.py

```diff
@@ -0,0 +1,118 @@
+import glob
+import subprocess
+import os
+import textwrap 
+import shutil
+
+path_scripts_2 = '../docs/source/scripts_2'
+if os.path.exists(path_scripts_2):
+    shutil.rmtree(path_scripts_2) 
+
+for dir_name in glob.glob("../autometa/**/*.py", recursive=True): 
+    if "__" not in dir_name: #to not include __init__ files
+        print(dir_name)
+        with open (dir_name) as fh_read:
+            path_temp_write = '../docs/temp_write.py'
+            fh_write = open(path_temp_write, 'w+')            
+            for line in fh_read:
+                if line.strip() == 'import argparse':
+                    line = line.strip()
+                    fh_write.write(line + '\n')
+                    break
+            # Reads text until the end of the block:
+            for line in fh_read:  # This keeps reading the file
+                if line.strip() == 'args = parser.parse_args()':
+                    line = line.strip()
+                    fh_write.write(line + '\n')
+                    break
+                line = line.strip()
+                fh_write.write(line + '\n')
+            fh_write.close()
+        
+        basename = os.path.basename(dir_name) #Extract the filename, eg: kmers.py
+        #print (basename)
+        len_base = len(basename) #needed to get the number of '=' in the header
+        file_name_base = os.path.splitext(basename)[0] #extract the kmers of kmers.py
+        file_name = file_name_base + ".rst" # make it kmers.rst   
+        #print(file_name)
+        main_path = os.path.dirname(os.path.abspath(dir_name)) # extract '/home/siddharth/Autometa/autometa/common/external'
+        #print(main_path)
+        path = main_path.split('/')
+        #print(main_path)
+        main_path = path [-1] # extraxt 'external'
+        path[path.index('autometa')] = 'docs/source/scripts_2' #replace autometa with the specified path in the list
+        path_ori=path='/'.join(path)
+
+        if not os.path.exists(path): #path now has 'docs/source/scripts_2/comon/external' instead of autometa
+            #make a new directory as per the structure in autometa
+            os.makedirs(path) #take care now, this will make each folder start with a lower case letter
+        path = os.path.join(path, file_name) #adding the kmers.rst to the above complete path
+        #now the path is 'docs/source/scripts_2/comon/external/samtools.rst'
+        #print(path)
+
+        cmd = 'python3 '+  os.path.abspath(path_temp_write) + ' -h'
+        #print(cmd)
+
+        path_temp_write_rst = os.path.abspath(path_temp_write)
```

Comment:
You have used `os.path.abspath(path_temp_write)` in line 53 as well. You can assign this to a variable sooner to reduce redundancy.